File: orangecandle-server/src/main/java/com/orangecandle/repository/Department.java
package com.orangecandle.repository;

import org.springframework.data.jpa.repository.JpaRepository;

import com.orangecandle.domain.Faculty;

public interface Department extends
		JpaRepository<com.orangecandle.domain.Department, Long> {

	public com.orangecandle.domain.Department findByNameAndFaculty(String name,
			Faculty faculty);

}

File: orangecandle-client/app/model/Department.js
Ext.define('OrangeCandle.model.Department', {
	extend : 'Ext.data.Model',
	config : {
		fields : [ {
			name : 'name',
			type : 'string'
		}, {
			name : 'facultyname',
			mapping : 'faculty.name'
		}, {
			name : 'schoolname',
			mapping : 'faculty.school.name'
		} ]
	}
});